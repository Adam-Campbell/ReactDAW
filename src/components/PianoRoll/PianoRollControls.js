import React from 'react';
import PropTypes from 'prop-types';
import SelectInput from '../SelectInput';

const PianoRollControls = props => (
    <div className="piano-roll-controls-container">
        <SelectInput 
            value={props.quantizeValue}
            handleChange={props.updateQuantizeValue}
            inputId="piano-roll-quantize-select"
            label="Select quantize level:"
            options={[
                { value: '32t', text: '32t' },
                { value: '32n', text: '32n' },
                { value: '16t', text: '16t' },
                { value: '16n', text: '16n' },
                { value: '8t', text: '8t' },
                { value: '8n', text: '8n' },
                { value: '4t', text: '4t' },
                { value: '4n', text: '4n' },
                { value: '2t', text: '2t' },
                { value: '2n', text: '2n' },
                { value: '1m', text: '1m' }
            ]}
        />
        <SelectInput 
            value={props.durationValue}
            handleChange={props.updateDurationValue}
            inputId="piano-roll-duration-select"
            label="Select note duration:"
            options={[
                { value: '32t', text: '32t' },
                { value: '32n', text: '32n' },
                { value: '16t', text: '16t' },
                { value: '16n', text: '16n' },
                { value: '8t', text: '8t' },
                { value: '8n', text: '8n' },
                { value: '4t', text: '4t' },
                { value: '4n', text: '4n' },
                { value: '2t', text: '2t' },
                { value: '2n', text: '2n' },
                { value: '1m', text: '1m' }
            ]}
        />
        <SelectInput 
            value={props.cursorValue}
            handleChange={props.updateCursorValue}
            inputId="piano-roll-cursor-select"
            label="Select tool:"
            options={[
                { value: 'pointer', text: 'Pointer' },
                { value: 'pencil', text: 'Pencil' }
            ]}
        /> 
    </div>
);

PianoRollControls.propTypes = {
    quantizeValue: PropTypes.string.isRequired,
    updateQuantizeValue: PropTypes.func.isRequired,
    durationValue: PropTypes.string.isRequired,
    updateDurationValue: PropTypes.func.isRequired,
    cursorValue: PropTypes.string.isRequired,
    updateCursorValue: PropTypes.func.isRequired
};

export default PianoRollControls;