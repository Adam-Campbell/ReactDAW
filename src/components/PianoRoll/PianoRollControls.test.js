import React from 'react';
import PianoRollControls from './PianoRollControls';
import { shallow } from 'enzyme';

test('renders correctly', () => {
    const component = shallow(
        <PianoRollControls 
            quantizeValue={'16n'}
            updateQuantizeValue={jest.fn()}
            durationValue={'16n'}
            updateDurationValue={jest.fn()}
            cursorValue={'pointer'}
            updateCursorValue={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
}); 