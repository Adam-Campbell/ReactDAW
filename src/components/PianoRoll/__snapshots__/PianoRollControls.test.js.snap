// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="piano-roll__controls-container"
>
  <SelectInput
    handleChange={[MockFunction]}
    inputId="piano-roll-quantize-select"
    label="Select quantize level:"
    options={
      Array [
        Object {
          "text": "32t",
          "value": "32t",
        },
        Object {
          "text": "32n",
          "value": "32n",
        },
        Object {
          "text": "16t",
          "value": "16t",
        },
        Object {
          "text": "16n",
          "value": "16n",
        },
        Object {
          "text": "8t",
          "value": "8t",
        },
        Object {
          "text": "8n",
          "value": "8n",
        },
        Object {
          "text": "4t",
          "value": "4t",
        },
        Object {
          "text": "4n",
          "value": "4n",
        },
        Object {
          "text": "2t",
          "value": "2t",
        },
        Object {
          "text": "2n",
          "value": "2n",
        },
        Object {
          "text": "1m",
          "value": "1m",
        },
      ]
    }
    value="16n"
  />
  <SelectInput
    handleChange={[MockFunction]}
    inputId="piano-roll-duration-select"
    label="Select note duration:"
    options={
      Array [
        Object {
          "text": "32t",
          "value": "32t",
        },
        Object {
          "text": "32n",
          "value": "32n",
        },
        Object {
          "text": "16t",
          "value": "16t",
        },
        Object {
          "text": "16n",
          "value": "16n",
        },
        Object {
          "text": "8t",
          "value": "8t",
        },
        Object {
          "text": "8n",
          "value": "8n",
        },
        Object {
          "text": "4t",
          "value": "4t",
        },
        Object {
          "text": "4n",
          "value": "4n",
        },
        Object {
          "text": "2t",
          "value": "2t",
        },
        Object {
          "text": "2n",
          "value": "2n",
        },
        Object {
          "text": "1m",
          "value": "1m",
        },
      ]
    }
    value="16n"
  />
  <SelectInput
    handleChange={[MockFunction]}
    inputId="piano-roll-cursor-select"
    label="Select tool:"
    options={
      Array [
        Object {
          "text": "Pointer",
          "value": "pointer",
        },
        Object {
          "text": "Pencil",
          "value": "pencil",
        },
      ]
    }
    value="pointer"
  />
</div>
`;
